indico/modules/rb/client/js/components/BookingBootstrapForm.jsx

```diff
@@ -379,28 +379,24 @@ class BookingBootstrapForm extends React.Component {
         )}
         {['every', 'daily'].includes(type) && (
           <Form.Group inline>
-            <DateRangePicker
-              startDate={startDate}
-              endDate={endDate}
-              onDatesChange={({startDate: sd, endDate: ed}) => this.updateDates(sd, ed)}
-              isOutsideRange={dt => {
-                return !isBookingStartDateValid(dt, isAdminOverrideEnabled, bookingGracePeriod);
-              }}
-              withFullScreenPortal={window.innerHeight < 730}
+            <DateRangePicker2
```

Comment:
After continuing from the form on the room booking splash page we show this page, where we allow changing the dates etc. However, we do not have a field there, clicking the date in the toolbar opens only the date(range) picker:

![image](https://github.com/user-attachments/assets/0441c326-1210-45db-9791-b631503e9996)

Is this something we can do with the new one? As far as the behavior is concerned, it opens the date range picker as if the calendar button in the start date field was clicked. This is in the `DateForm` component (it only renders when it's open, so it would need to show the date picker immediately)

Reply:
So you want the whole thing to behave as a big click target? Is there a reason why it's not ok to use the same date range picker as in the forms?

Reply:
The click target is only the part with the date. "Daily" and the times open different popovers (where one can then change only those parts):

![image](https://github.com/user-attachments/assets/2e8f5047-7e16-4536-aaa1-3e7bd58a989d)
![image](https://github.com/user-attachments/assets/19266fb0-d2f9-45da-a586-c6675d39922f)
![image](https://github.com/user-attachments/assets/6d16b389-7061-4a91-bf68-0f258983b648)


---

I don't think it would look good here to just show what's basically a form field in that toolbar (and honestly, I don't really want to do any major refactoring on the BookingFilterBar thingy)...

~~Showing the field w/ a separate click needed to get to the calendar inside the popover would look weird and not be very user-friendly.~~

OTOH (and I started writing that part of the comment only after taking the other screenshots, hence the contradiction to my previous paragraph): The time picker also shows a form field. So I guess maybe it would be *acceptable* to show the normal Date(Range)Picker in there, since it would be pretty similar to what we have for the time picker as well...

---

Reply:
Maybe we can cheat a little with styling to make it look like it's not an input until you click on it.

Reply:
I think this would most likely get into the "(and honestly, I don't really want to do any major refactoring on the BookingFilterBar thingy)" territory ;)

(currently the picker is not even rendered until you click in the toolbar)

Reply:
Ok, I'll check it out and see what we can do.

Reply:
I've poked at this a little. It actually looks like you need a stand-alone calendar that can be inserted into any random popup. Which we basically already have, just not a React wrapper.

Reply:
So this stand-alone calendar React component would use `<ind-calendar>` to show one or two date grids, and an "Apply" button so that you confirm the edit. Off the top of my head, ideally this would be a drop-in replacement for `DateForm`. Perhaps internally it would need two separate components, one for single date, and one for date range.